cupy/random/cupy_distributions.cu

```diff
@@ -12,6 +12,11 @@
 
 struct rk_state {
 
+#ifdef CUPY_USE_BINOMIAL
+    int has_binomial;
+    int nsave, m;
+    double psave, r, q, fm, p1, xm, xl, xr, c, laml, lamr, p2, p3, p4;
```

Comment:
To be honest, i'd rather have this in a different struct rather than rk_state

Reply:
Yeah writing this in a separate struct will increase readability.

Reply:
I am writing separate class for this, but while doing so I have a doubt. How to use the multiple structs in generator dispatcher.

Reply:
You can use a ndarray that has the same size in bytes as the size of the other struct to keep the additional state. The ndarray's type can be `uint8`.

Reply:
https://github.com/cupy/cupy/blob/995ce208dc475525c37b6a449e64d7e40bcb5fdc/cupy/random/_bit_generator.pyx#L77

Yes, this is what we do to allocate the state, you can do this for the binomial and pass the pointer as a parameter to the distribution